src/Language/Futhark/Interpreter.hs

```diff
@@ -1626,13 +1626,14 @@ initialCtx =
       fun2t $ \i xs -> do
         let (shape, xs') = fromArray xs
         return $
-          if asInt i > 0
-            then
-              let (bef, aft) = splitAt (asInt i) xs'
-               in toArray shape $ aft ++ bef
-            else
-              let (bef, aft) = splitFromEnd (- asInt i) xs'
-               in toArray shape $ aft ++ bef
+          let idx = rem (asInt i) (length xs')
```

Comment:
What if the array is empty?

Please also add a test case for this.

Reply:
Good catch